File: problem_solving_for_beginners/python/perfect_number.py
import math

def isPerfect(n):
    if n < 1:
        print("Enter a positive number")
    else:
        sqr = int(math.sqrt(n)) + 1
        res = 1
        for i in range (2, sqr):
            if n % i == 0:
                tmp = n // i
                res += i + tmp
        return res


n = int(input("Enter a positive number: "))

res = isPerfect(n)

if res == n:
    print("Yes")
else:
    print("No")

File: problem_solving_for_beginners/Number_Problems/c_programming/perfect_number.c
#include <stdio.h>
#include <math.h>

int isperfect(int num)
{
    int  i, tmp, sqr, res = 1;
     sqr = sqrt(num);
    for (i = 2; i <= sqr; i++)
    {
        if (num % i == 0)
        {
            tmp = num / i;
            res = res + i + tmp;
        }
    }
    return (res);
}

int main(void)
{
    int n, res;
    printf("Enter a positive number: ");
    scanf("%d", &n);

    res = isperfect(n);
    if (res == n)
        printf("Yes\n");
    else
        printf("No\n");

    return 0;
}